### PR TITLE
fix: conditionally add "needs testing" label based on existing labels

### LINKED ISSUE #14470: needs testing label repeated even when Tested label is present

### PR DESCRIPTION
**Pull request by areebahmeddd**
## Proposed Changes

If a PR has Tested label, it won't add another label "needs-testing" even if you re request review

Fixes: #14470
cc: @yash-learner

Tagging: @ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate the bug or test the new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is placed in I18n files.
- [ ] Prepare a screenshot or demo video for the changelog entry and attach it to the issue.
- [x] Request peer reviews.
- [ ] Complete QA on mobile devices.
- [ ] Complete QA on desktop devices.
- [ ] Add or update Playwright tests for related changes

## Summary by CodeRabbit

* **Chores**
  * PR automation refined: skips bot reviewers when re-requesting review.
  * Label handling made conditional: "needs testing" and "needs review" are added only if not already present.
  * Still removes "changes required" when appropriate.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by nihal467** [comment]
marking it as tested to avoid bot

**Review by yash-learner — CHANGES_REQUESTED** [review]
-> Please file issue and link to the PR
-> Name your branch names meaningful from next time
-> where is case for not to add `needs review` and remove `changes requested` when copilot review gets automatically triggered as mentioned here -> https://github.com/ohcnetwork/care_fe/pull/14268#issuecomment-3511582142 ?

File: .github/workflows/pr-automation.yml [hunk 1]
```diff
@@ -72,12 +72,18 @@ jobs:
...
 
       - name: Handle Review Re-Request
-        if: github.event.action == 'review_requested'
+        if: github.event.action == 'review_requested' && github.event.requested_reviewer.type != 'Bot'
         env:
           GH_TOKEN: ${{ steps.generate-token.outputs.token }}
           PR_NUMBER: ${{ github.event.pull_request.number }}
           REPO: ${{ github.repository }}
         run: |
+          HAS_TESTED=$(gh pr view $PR_NUMBER --repo $REPO --json labels --jq '.labels[].name' | grep -x "Tested" || echo "")
```
Comment (yash-learner) [inline .github/workflows/pr-automation.yml]: This doesn't check whether PR already has a `needs testing`, label I think this will not cause issue because Github api will take care -> does not allow duplicate labels, we will be making just an additional api call

**Review by yash-learner — APPROVED** [review]

File: .github/workflows/pr-automation.yml (hunk 1 above)
Comment (amjithtitus09) [reply .github/workflows/pr-automation.yml]: @areebahmeddd same for `needs review`

**Review by amjithtitus09 — CHANGES_REQUESTED** [review]

**Review by yash-learner — APPROVED** [review]
LGTM

**Review by amjithtitus09 — APPROVED** [review]
